packages/hoppscotch-app/newstore/WebSocketSession.ts

```diff
@@ -0,0 +1,275 @@
+import { pluck, distinctUntilChanged } from "rxjs/operators"
+import DispatchingStore, { defineDispatchers } from "./DispatchingStore"
+import {
+  HoppRealtimeLog,
+  HoppRealtimeLogLine,
+} from "~/helpers/types/HoppRealtimeLog"
+
+type HoppWSProtocol = {
+  value: string
+  active: boolean
+}
+
+type HoppWSRequest = {
+  endpoint: string
+  protocols: HoppWSProtocol[]
+}
+
+export type HoppWSSession = {
+  request: HoppWSRequest
+  connectingState: boolean
+  connectionState: boolean
+  log: HoppRealtimeLog
+  socket: WebSocket
```

Comment:
We cannot persist runtime values!

Reply:
sorry for the late response.

runtime values are **just** saved in the session, not the disk.

the current state of the realtime sockets that if the socket is running then, if you switched tabs and came back every thing will look normal but the socket will keep running in the background and will not close.

if I did not save a reference to the socket then when the component is recreated it will redefine the socket variable an the old socket is unreachable.

if we did not want to save these values then, we will have to close the socket which defeats the purpose of the whole thing. #1196

may I ask if you have any alternative approach.

Reply:
Oh okay, cool, but the not saving to disk part, is it explicitly mentioned somewhere in the code (`localpersistence` should be the area for that if it is there). If we are relying on JSON parsing behaviours instead to prevent the write, then it would be cool if we had that explicitly defined, we have had a lot of instances where the spec implementation vary across browsers or libraries might break some side behaviours, so explicitly defining in localstorage what gets saved would be nice.

If we do have such code explicitly specified then, just can you point me to that line ? 😅